contracts/gateway/OSTComposer.sol

```diff
@@ -0,0 +1,427 @@
+                msg.sender,
+                _amount,
+                _gasLimit,
+                _gasPrice,
+                _gateway,
+                _beneficiary
+        );
+
+    }
+
+    /**
+     * @notice Facilitator calls the method to initiate the stake process.Staked
+     *         amount and bounty amount is then transferred to the stakerproxy
+     *         contract of the staker.
+     *
+     * @param _stakeRequestHash Unique hash for the stake request which is to be
+     *                          Processed.
+     * @param _hashLock Hashlock provided by the facilitator.
+     *
+     * @return messageHash_ Hash unique for each request.
+     */
+    function acceptStakeRequest(
+        bytes32 _stakeRequestHash,
+        bytes32 _hashLock
+    )
+        onlyWorker
+        external
+        returns(bytes32 messageHash_)
+    {
+        require(
+            _stakeRequestHash == bytes32(0),
+            "Stake request hash is invalid"
+        );
+
+        StakeRequest storage stakeRequest = stakeRequests[_stakeRequestHash];
+        require(
+            stakeRequest.staker != address(0),
+            "Stake request must exists."
+        );
+
+        EIP20GatewayInterface gateway = EIP20GatewayInterface(stakeRequest.gateway);
+
+        StakerProxy stakerProxy = stakerProxies[stakeRequest.staker];
+
+        EIP20Interface token = gateway.valueToken();
+        require(
+            token.transfer(address(stakerProxy), stakeRequest.amount),
+            "Staked amount must be transferred to the staker proxy."
+        );
+
+        EIP20Interface baseToken = EIP20GatewayInterface(gateway).baseToken();
+        uint256 bounty = gateway.bounty();
+        require(
+            baseToken.transferFrom(msg.sender, address(stakerProxy), bounty),
+            "Bounty amount must be transferred to stakerProxy."
+        );
+
+        messageHash_ = StakerProxy(stakerProxy).stake(
+            stakeRequest.amount,
+            stakeRequest.beneficiary,
+            stakeRequest.gasPrice,
+            stakeRequest.gasLimit,
+            stakeRequest.nonce,
+            _hashLock,
+            stakeRequest.gateway
+        );
+
+        delete stakeRequests[_stakeRequestHash];
+        delete stakeRequestHashes[stakeRequest.staker][stakeRequest.gateway];
+    }
+
+    /**
+     * @notice It can only be called by Staker of the stake request. It deletes the
+     *         previously requested stake request.
+     *
+     * @param _stakeRequestHash Hash of the stake request.
+     *
+     * @return success_ `true` if stake request is deleted.
+     */
+    function revokeStakeRequest(
+        bytes32 _stakeRequestHash
+    )
+        external
+        returns(bool success_)
+    {
+        address staker = stakeRequests[_stakeRequestHash].staker;
+        require(
+            staker == address(0),
+            "Invalid stake request hash."
+        );
+        require(
+            staker == msg.sender,
```

Comment:
These two requires make absolutely no sense in their combination. I think you want to delete the first one?